File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/inss/sobresalarios/LegendaDaFormulaDoInss.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios;

import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.Inss;
import br.jus.trt8.pjecalc.negocio.dominio.historicosalarial.HistoricoSalarial;
import br.jus.trt8.pjecalc.negocio.dominio.verbacalculo.VerbaDeCalculo;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class LegendaDaFormulaDoInss
implements Serializable {
    private static final long serialVersionUID = -750724924346677245L;
    private Inss inss;
    private List<String> basesHistoricos;
    private List<String> basesVerbas;

    public LegendaDaFormulaDoInss(Inss inss) {
        this.inss = inss;
        this.basesHistoricos = new ArrayList<String>();
        this.basesVerbas = new ArrayList<String>();
        this.init();
    }

    private void init() {
        for (HistoricoSalarial historico : this.inss.getCalculo().getHistoricosSalariais()) {
            if (!historico.getIncidenciaINSS().booleanValue()) continue;
            this.basesHistoricos.add(historico.getNome());
        }
        for (VerbaDeCalculo verba : this.inss.getCalculo().getVerbas()) {
            if (!verba.getAtivo().booleanValue() || !verba.getIncidenciaINSS().booleanValue()) continue;
            this.basesVerbas.add(verba.getNome());
        }
    }

    public List<String> getBasesHistoricos() {
        return this.basesHistoricos;
    }

    public List<String> getBasesVerbas() {
        return this.basesVerbas;
    }

    public String getLegendaSalariosDevidos() {
        StringBuilder str = new StringBuilder();
        if (!this.basesVerbas.isEmpty()) {
            for (String nomeDaVerba : this.basesVerbas) {
                if (this.basesVerbas.indexOf(nomeDaVerba) == 0) {
                    str.append(nomeDaVerba);
                    continue;
                }
                str.append(" + ").append(nomeDaVerba);
            }
        }
        return str.toString().toUpperCase();
    }

    public String getLegendaSalariosPagos() {
        StringBuilder str = new StringBuilder();
        if (!this.basesHistoricos.isEmpty()) {
            for (String nomeDaVerba : this.basesHistoricos) {
                if (this.basesHistoricos.indexOf(nomeDaVerba) == 0) {
                    str.append(nomeDaVerba);
                    continue;
                }
                str.append(" + ").append(nomeDaVerba);
            }
        }
        return str.toString().toUpperCase();
    }
}

File: src/lib/pjecalc/core/dominio/calculo/inss/sobresalarios/legenda-da-formula-do-inss.ts
/**
 * PJe-Calc v2.15.1 — LegendaDaFormulaDoInss
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.LegendaDaFormulaDoInss
 *
 * Ref Java: pjecalc-fonte/.../sobresalarios/LegendaDaFormulaDoInss.java
 *
 * Helper que monta strings descritivas das bases de INSS de um cálculo:
 *   - basesHistoricos: nomes de HistoricoSalarial com incidencia INSS
 *   - basesVerbas: nomes de VerbaDeCalculo ativa com incidencia INSS
 * getLegendaSalariosDevidos / getLegendaSalariosPagos concatenam os nomes
 * com " + " em maiúsculas.
 */
import type { Inss } from '../inss';

export class LegendaDaFormulaDoInss {
  private inss: Inss;
  private basesHistoricos: string[] = [];
  private basesVerbas: string[] = [];

  constructor(inss: Inss) {
    this.inss = inss;
    this.init();
  }

  private init(): void {
    const calc = this.inss.getCalculo();
    if (!calc) return;
    const c = calc as unknown as {
      getHistoricosSalariais?: () => Iterable<{ getIncidenciaINSS?: () => boolean; getNome?: () => string }>;
      getVerbas?: () => Iterable<{ getAtivo?: () => boolean; getIncidenciaINSS?: () => boolean; getNome?: () => string }>;
    };
    for (const historico of c.getHistoricosSalariais?.() ?? []) {
      if (!historico.getIncidenciaINSS?.()) continue;
      const nome = historico.getNome?.();
      if (nome) this.basesHistoricos.push(nome);
    }
    for (const verba of c.getVerbas?.() ?? []) {
      if (!verba.getAtivo?.()) continue;
      if (!verba.getIncidenciaINSS?.()) continue;
      const nome = verba.getNome?.();
      if (nome) this.basesVerbas.push(nome);
    }
  }

  getBasesHistoricos(): string[] { return this.basesHistoricos; }
  getBasesVerbas(): string[] { return this.basesVerbas; }

  getLegendaSalariosDevidos(): string { return this.basesVerbas.join(' + ').toUpperCase(); }
  getLegendaSalariosPagos(): string { return this.basesHistoricos.join(' + ').toUpperCase(); }
}
